tests/test_vonkarman.py

```diff
@@ -0,0 +1,209 @@
+# Copyright (c) 2012-2017 by the GalSim developers team on GitHub
+# https://github.com/GalSim-developers
+#
+# This file is part of GalSim: The modular galaxy image simulation toolkit.
+# https://github.com/GalSim-developers/GalSim
+#
+# GalSim is free software: redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the following
+# conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions, and the disclaimer given in the accompanying LICENSE
+#    file.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions, and the disclaimer given in the documentation
+#    and/or other materials provided with the distribution.
+#
+
+from __future__ import print_function
+import numpy as np
+import os
+import sys
+
+from galsim_test_helpers import *
+
+try:
+    import galsim
+except ImportError:
+    sys.path.append(os.path.abspath(os.path.join(path, "..")))
+    import galsim
+
+
+@timer
+def test_vk(slow=False):
+    """Test the generation of VonKarman profiles
+    """
+    if __name__ == '__main__' and slow:
+        lams = [300.0, 500.0, 1100.0]
+        r0_500s = [0.05, 0.15, 0.3]
+        L0s = [1e10, 25.0, 10.0]
+        do_deltas = [False, True]
```

Comment:
How slow is this?  We have some pretty slow tests already in the name==main branches.  Even if it's a few minutes, I think we should probably remove the `slow` option and just always do this when run from the command line. 

One benefit of this is that `run_all_tests` will hit these slow tests, which is something I tend to run before releases to make sure even the weird stuff hasn't broken at some point along the way during development.

Reply:
Thanks for asking; I think I actually must not have run all of these since adding the `do_shoot` and `do_kvalue` tests, since these failed on my laptop.  As far as I can tell, the problem boils down to having an extreme maxk/stepk ratio for a vonKarman that includes a delta function component.  (The maxk in particular become infinite).  Since the tests pass (after some minor tweaks to `gsparams.maximum_fft_size` and `do_shoot`) when `do_delta=False` though, I'm satisfied things are okay, and opted to just skip these when `do_delta=True` (the c++ part of the profile is the same in either case).

These are pretty slow, about 13 minutes.  Removing the `args.slow` is your call; I'm happy either way.

Reply:
I picked up your gauntlet and worked on speeding up the code some.  

Now running test_vonkarman via nosetests takes 6.6 seconds for the whole thing (down from 91 seconds).  

And running directly with slow=True takes 330 seconds (down from 1900 sec for me).  I think that's fast enough now to include by default for command-line runs, so I switched it.

Most of the rest of the slow parts are calling kValue (via drawKImage).  I feel like there's probably something we could do with the large k asymptotic form for BesselK, but I didn't try to work out the algebra with that to see if there is some speed up possible there.  Maybe not, since there is still an exp inside the exp, which doesn't sound like it would simplify nicely, although it's possible.

Reply:
Thanks Mike, this is really nice!  Do you recall what the biggest improvement was?